### DISCUSSION TITLE
Firmware settings using Gcode.

### DISCUSSION THREAD

**Discussion by infamous-panda** [terjeio/grblHAL#347] [General]
Is there a way to change GRBL firmware settings within a Gcode program? An example would be a job using both a router and laser engraver where $32 laser mode can be changed mid job. Also interested in changing acceleration values for specific jobs. 

Iosender seems to ignore $ commands in Gcode. Is this a firmware or sender limitation?

**Comment by rvalotta** [comment]
I would like to second a M-Code to switch between laser and Spindle.  Setting $32=1 is a flash write and I would prefer to limit what my "Default Config" is and just switch between laser mode and spindle mode

**Reply by terjeio** [reply]
> Iosender seems to ignore $ commands in Gcode. Is this a firmware or sender limitation?

It is a sender limitation. Perhaps I should allow $ commands, I have to think about that as $ commands in gcode programs may result in an error beeing raised.

> I would like to second a M-Code to switch between laser and Spindle. 

Does LinuxCNC have a definition for one? I am not too keen to add non-standard M-codes to the core, but adding them to plugin code is ok.

Please note that this repo is going to be archived soon.

**Comment by terjeio** [comment]
>  An example would be a job using both a router and laser engraver where $32 laser mode can be changed mid job.

A [mini plugin](https://github.com/grblHAL/Plugins_spindle/blob/master/select.c) for switching between Modbus VDF spindle and PWM output has been added to the [Spindle plugin](https://github.com/grblHAL/Plugins_spindle), currently it uses `M104` to switch between them: `M104P0` for PWM (laser) and `M104P1` for VFD.
Available when `#define DUAL_SPINDLE` is uncommented in _my_machine.h_ for drivers that has support,

> Also interested in changing acceleration values for specific jobs.

The [OpenPNP plugin](https://github.com/grblHAL/Plugin_OpenPNP) adds a M-code for that.
